ja/core-libraries/file-folder.rst

```diff
@@ -3,13 +3,445 @@ Folder & File
+
+        $dir = new Folder(WWW_ROOT);
+        $files = $dir->read(true, ['files', 'index.php']);
+        /*
+        Array
+        (
+            [0] => Array // Folders
+                (
+                    [0] => css
+                    [1] => img
+                    [2] => js
+                )
+            [1] => Array // Files
+                (
+                    [0] => .htaccess
+                    [1] => favicon.ico
+                    [2] => test.php
+                )
+        )
+        */
+
+.. php:method:: realpath(string $path)
+
+    引数のパス内にある ".." の名前を解決したパスを返します。
+
+.. php:staticmethod:: slashTerm(string $path)
+
+    引数の $path に終端のスラッシュを付けたパスを返します。
+
+.. php:method:: tree(null|string $path = null, array|boolean $exceptions = true, null|string $type = null)
+
+    ディレクトリ一覧とその中のファイル一覧を返します。
+
+File API
+========
+
+.. php:class:: File(string $path, boolean $create = false, integer $mode = 755)
+
+::
+
+    // 0644 のパーミッションで新しいファイルを作成します
+    $file = new File('/path/to/file.php', true, 0644);
+
+.. php:attr:: Folder
+
+    ファイルが属するフォルダ・オブジェクト。
+
+.. php:attr:: name
+
+    拡張子付きのファイル名。
+    拡張子なしのファイル名を返す :php:meth:`File::name()` とは異なります。
+
+.. php:attr:: info
+
+    ファイル情報の配列。
+    代わりに :php:meth:`File::info()` を使用してください。
+
+.. php:attr:: handle
+
+    ファイルをオープンしている場合のファイルハンドラを保持します。
+
+.. php:attr:: lock
+
+    ファイルの読み書き時のロックを有効にします。
+
+.. php:attr:: path
+
+    現在のファイルの絶対パス。
+
+.. php:method:: append(string $data, boolean $force = false)
+
+    引数の文字列をファイルへ追記します。
+
+.. php:method:: close()
+
+    ファイルがオープンされていた場合、そのファイルをクローズします。
+
+.. php:method:: copy(string $dest, boolean $overwrite = true)
+
+    ファイルを $dest へコピーします。
+
+.. php:method:: create()
+
+    ファイルを作成します。
+
+.. php:method:: delete()
+
+    ファイルを削除します。
+
+.. php:method:: executable()
+
+    ファイルに実行権限が付いていた場合に ``true`` を返します。
+
+.. php:method:: exists()
+
+    ファイルが存在した場合に ``true`` を返します。
+
+.. php:method:: ext()
+
+    ファイルの拡張子を返します。
+
+.. php:method:: Folder()
+
+    現在のフォルダを返します。
+
+.. php:method:: group()
+
+    ファイルのグループを返します。エラーの場合は ``false`` を返します。
+
+.. php:method:: info()
+
+    ファイル情報を返します。
+
+.. php:method:: lastAccess( )
+
+    最新のアクセス時間を返します。
+
+.. php:method:: lastChange()
+
+   最新の更新時間を返します。エラーの場合は ``false`` を返します。
+
+.. php:method:: md5(integer|boolean $maxsize = 5)
+
+    ファイルサイズを事前にチェックした上で、ファイルの md5 チェックサムを返します。
+    エラーの場合、 ``false`` を返します。
+
+.. php:method:: name()
+
+    拡張子を省いたファイル名を返します。
+
+.. php:method:: offset(integer|boolean $offset = false, integer $seek = 0)
+
+    現在オープンしているファイルのオフセット値を設定または取得します。
+
+.. php:method:: open(string $mode = 'r', boolean $force = false)
+
+    現在のファイルを引数の $mode でオープンします。
+
+.. php:method:: owner()
+
+    ファイルのオーナーを返します。
+
+.. php:method:: perms()
+
+    ファイルの "chmod" (パーミッション) を返します。
+
+.. php:staticmethod:: prepare(string $data, boolean $forceWindows = false)
+
+    ASCII 文字列をファイルへ書き出す事前処理を行います。
+    現在の実行環境に合わせて改行文字を変換します。
+    Windows なら "\\r\\n" を、その他の環境なら "\\n" が利用されます。
```

Comment:
Can you fix other language versions too? It seems we have to escape backslashes.